src/Cheque/send_token_by_identifier.md

```diff
@@ -88,14 +92,16 @@ In such case where both are attestations, the "subject" of the 1st attestation,
+We note that despite having described the protocol using general multiplication group notation, the implementations will be based on elliptic curves. Therefore, *G, V, u, v* will be points on an elliptic curve where *u* and *v* are computed using generator points *G* and *V*. Furthermore, this also means that the computation in step 4 for Bob and step 4 for the smart contract will happen over the integers, modulo the curve order. 
+We note that it is crucial that the generators *G* and *V* are constructed such that they work over the same (large) subgroup. I.e. that *V∈ &lt;G&gt;* and that **no-one** knows the discrete log of *V* to base *G* (and vice versa). Otherwise breaks become trivial.
+Furthermore it is also crucial that the curve has co-factor 1, or if not, that the point *G* generates a subgroup of large prime order.
 
 [This post](https://crypto.stackexchange.com/questions/34863/ec-schnorr-signature-multiple-standard) mentions some standards for EC-based Fiat-Shamir Schnorr proofs and thus where to look for further details.
 
 ## In the case of using a JavaScript deployed as a service
 
-Furthermore, we note that there does not seem to be standard Javascript libraries to compute such an elliptic curve Fiat-Shamir Schnorr proof. Thus this could be allowed to be supported by a third party (specifically step 1-3 for Bob). However, if such a library is malicious, it will learn *x* and thus be able to impersonate Bob. This *must* not happen. Thus instead of constructing a proof of knowledge of *x* s.t. *s'=s<sup>x</sup>* Bob uses such a library to construct a proof of knowledge of *x+w* s.t. *s'=<sup>x+w</sup>* for a random *w*. Based on this Bob will instead send *(s, s', t, d, w)* in step 4 and the server will instead verify *s<sup>d</sup>=t·s'<sup>c·w</sup>* in step 3.
+Furthermore, we note that there does not seem to be standard Javascript libraries to compute such an elliptic curve Fiat-Shamir Schnorr proof. Thus this could be allowed to be supported by a third party (specifically step 1-3 for Bob). However, if such a library is malicious, it will learn *x* and thus be able to impersonate Bob. This *must* not happen. Thus instead of constructing a proof of knowledge of *x* Bob uses such a library to construct a proof of knowledge of *x+w* for a random *w*. Based on this Bob will instead send *(V, s, t, d, w)* in step 3 and the server will instead verify *V<sup>d</sup>=t·s<sup>c·w</sup>*.
+(A similar trick must be done for the proof used to redeem the cheque using the smart contract.)
 Still, even this approach does allow for a front-running displacement attack in case the Javascript library sends the query to its owners who also do mining, and so the miner will learn *x* and thus be able to impersonate Bob once he tried to cash the cheque.
```

Comment:
JavaScript library sends the query to its author (not owner - for clarity that you refer to the author not the person who owns the copy of code)

Reply:
Great, thanks! I also realised we have actually not talked with Oleg about this yet, so I think we maybe should mention that we would like to include this to prevent malicious java script libraries. Although his current implementation does everything locally as far as I can see.

Reply:
Depends on if he need to change this in case some other implementation of this protocol needs to use 3rd party library - for his own implementation he uses the raw elliptic curve math we created anyway.